behat/features/bootstrap/FeatureContext.php

```diff
@@ -67,4 +67,138 @@ public function iVisitNodePageOfType($title, $type) {
+      ->propertyCondition('title', $title)
+      ->propertyCondition('status', NODE_PUBLISHED)
+      ->range(0, 1)
+      ->execute();
+    if (empty($result['node'])) {
+      $params = array(
+        '@title' => $title,
+        '@type' => $type,
+      );
+      throw new \Exception(format_string("Node @title of @type not found.", $params));
+    }
+    $nid = key($result['node']);
+    if (node_access('update', node_load($nid), user_load_by_name($this->user->name))) {
+      return;
+    }
+    $params = array(
+      '@title' => $title,
+      '@type' => $type,
+    );
+    throw new \Exception(format_string("You can't edit the node @title of @type not found.", $params));
+  }
+
+  /**
+   * @When I start creating node of type :type
+   */
+  public function iStartCreatingNodeOfType($type) {
+    $this->getSession()->visit($this->locatePath('node/add/' . $type));
+  }
+
+  /**
+   * @When I create :title node of type :type
+   */
+  public function iCreateNodeOfType($title, $type) {
+    $account = user_load_by_name($this->user->name);
+    $values = array(
+      'type' => $type,
+      'uid' => $account->uid,
+      'status' => 1,
+      'comment' => 1,
+      'promote' => 0,
+    );
+    $entity = entity_create('node', $values);
+    $wrapper = entity_metadata_wrapper('node', $entity);
+    $wrapper->title->set($title);
+    $wrapper->field_github_id->set(123456);
+    $wrapper->save();
+  }
+
+  /**
+   * @When I delete :arg1 node of type :arg2
+   */
+  public function iDeleteNodeOfType($title, $type) {
+    $query = new \entityFieldQuery();
+    $result = $query
+      ->entityCondition('entity_type', 'node')
+      ->entityCondition('bundle', strtolower($type))
+      ->propertyCondition('title', $title)
+      ->propertyCondition('status', NODE_PUBLISHED)
+      ->range(0, 1)
+      ->execute();
+    if (empty($result['node'])) {
+      $params = array(
+        '@title' => $title,
+        '@type' => $type,
+      );
+      throw new \Exception(format_string("Node @title of @type not found.", $params));
+    }
+    $nid = key($result['node']);
+    node_delete($nid);
+  }
+
+  /**
+   * @Then I should not be able to add content to :title repository
+   */
+  public function iShouldNotBeAbleToAddContentToRepository($title) {
+    $query = new \entityFieldQuery();
+    $result = $query
+      ->entityCondition('entity_type', 'node')
+      ->entityCondition('bundle', 'repository')
+      ->propertyCondition('title', $title)
+      ->propertyCondition('status', NODE_PUBLISHED)
+      ->range(0, 1)
+      ->execute();
+    if (empty($result['node'])) {
+      $params = array(
+        '@title' => $title,
+        '@type' => 'repository',
+      );
+      throw new \Exception(format_string("Node @title of @type not found.", $params));
+    }
+    $nid = key($result['node']);
+    $account = user_load_by_name($this->user->name);
+    if (og_user_access('node', $nid, 'access group', $account)) {
```

Comment:
this is actually wrong -- there's no such permission defined by og. it should be enough to node_access() the group.